neqo-client/src/main.rs

```diff
@@ -597,18 +598,22 @@ mod old {
     use neqo_common::{matches, Datagram};
     use neqo_crypto::{AuthenticationStatus, Cipher};
     use neqo_transport::{
-        Connection, ConnectionEvent, FixedConnectionIdManager, Output, State, StreamType,
+        Connection, ConnectionEvent, Error, FixedConnectionIdManager, Output, State, StreamType,
     };
 
     use super::{emit_datagram, get_output_file, Args};
 
     trait HandlerOld {
-        fn handle(&mut self, args: &Args, client: &mut Connection) -> Res<bool>;
+        fn handle(&mut self, client: &mut Connection) -> Res<bool>;
+    }
+
+    trait DownloadNext {
+        fn download_next(&mut self, client: &mut Connection) -> Res<bool>;
     }
```

Comment:
I don't think you need this trait, because you're only calling `download_next` on the `PostConnectHandlerOld` concrete type. Instead, create an impl block for `PostConnectHandlerOld`:
```
    impl<'a> PostConnectHandlerOld<'a> {
         fn download_next(blah) { }
    }
```
after its struct definition and put the method there.